owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt

```diff
@@ -1361,6 +1365,37 @@ class FileDisplayActivity : FileActivity(),
         }
     }
 
+    private fun openShortcutFileInBrowser(file: OCFile) {
+        val url = extractUrlFromFile(file.storagePath.toString())
+        val truncatedUrl = url?.truncateWithEllipsis(MAX_URL_LENGTH)
+        val message = SpannableStringBuilder()
+            .append(getString(R.string.open_shortcut_description))
+            .append(truncatedUrl)
+        val dialog = AlertDialog.Builder(this@FileDisplayActivity)
+            .setTitle(getString(R.string.open_shortcut_title))
+            .setMessage(message)
+            .setPositiveButton(R.string.drawer_open) { view, _ ->
+                url?.let {
+                    goToUrl(url)
+                }
+                view.dismiss()
+            }
+            .setNegativeButton(R.string.share_cancel_public_link_button) { view, _ ->
+                view.dismiss()
+            }
+            .setCancelable(true)
+            .create()
+        dialog.show()
+    }
+
+    private fun String.truncateWithEllipsis(maxLength: Int): String {
+        return if (this.length > maxLength) {
+            "${this.substring(0, maxLength)}..."
```

Comment:
I don't know if in this case it would be better to ellipsize the URL at the end or in the middle, let's take a look at similar apps and see how they display it